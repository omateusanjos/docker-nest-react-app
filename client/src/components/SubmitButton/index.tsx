import React from "react";

interface SubmitButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
}

const SubmitButton = ({ children, ...props }: SubmitButtonProps) => (
  <button
    className="text-logo-btn p-2 rounded-lg my-4 font-bold"
    type="submit"
    {...props}
  >
    {children}
  </button>
);

export { SubmitButton };
