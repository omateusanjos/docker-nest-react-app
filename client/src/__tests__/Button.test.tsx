import React from "react";
import { render, RenderResult } from "@testing-library/react";
import { SubmitButton } from "../components/SubmitButton/index";
import "@testing-library/jest-dom";

describe("\nSubmitButton component", () => {
  let wrapper: RenderResult | null;

  beforeEach(() => {
    wrapper = render(<SubmitButton>Submit</SubmitButton>);
    expect(wrapper).not.toBeNull();
  });

  afterEach(() => {
    wrapper = null;
  });

  it("should render a button", () => {
    expect(
      wrapper?.getByRole("button", {
        name: "Submit",
      })
    ).toBeInTheDocument();
  });

  it("should render a button and rerender but no text", () => {
    expect(wrapper?.getByRole("button")).toBeInTheDocument();
    wrapper?.rerender(<SubmitButton children={undefined} />);
    expect(wrapper?.getByRole("button")).toHaveTextContent("");
  });

  it("button native property assignment: disabled", () => {
    expect(wrapper?.getByRole("button")).not.toBeDisabled();
    wrapper?.rerender(<SubmitButton disabled children={undefined} />);
    expect(wrapper?.getByRole("button")).toHaveAttribute("disabled");
  });

  it("should match snapshot", () => {
    expect(wrapper?.container).toMatchSnapshot();
  });
});
