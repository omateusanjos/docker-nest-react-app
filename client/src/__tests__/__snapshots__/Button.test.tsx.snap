// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`
SubmitButton component should match snapshot 1`] = `
<div>
  <button
    class="text-logo-btn p-2 rounded-lg my-4 font-bold"
    type="submit"
  >
    Submit
  </button>
</div>
`;
